### PR TITLE
Implement drop constraint as AlterTableTask

### PR DESCRIPTION
**Pull request by mfussenegger**
Follows the approach introduced by:

- https://github.com/crate/crate/pull/15055
- https://github.com/crate/crate/pull/15188

Motivated by:

https://github.com/crate/crate/issues/11939

Reduces the amount of places where both indices and templates are
modified.

### CODE REVIEW

File: server/src/main/java/io/crate/metadata/doc/DocTableInfo.java [hunk 1]
```diff
@@ -733,6 +729,32 @@ private void updateParentsInnerTypes(ColumnIdent column,
         }
     }
 
+    public DocTableInfo dropConstraint(String constraint) {
+        List<CheckConstraint<Symbol>> newConstraints = checkConstraints.stream()
+            .filter(x -> !x.name().equals(constraint))
+            .toList();
```
Comment (jeeminso) [inline server/src/main/java/io/crate/metadata/doc/DocTableInfo.java:735]: Minor: doesn't filter go through all elements even when the one is found? I think we only need to remove the first element with the name.

File: server/src/main/java/io/crate/metadata/doc/DocTableInfo.java [hunk 2]
```diff
@@ -733,6 +729,32 @@ private void updateParentsInnerTypes(ColumnIdent column,
...
+        if (newConstraints.size() == checkConstraints.size()) {
+            return this;
+        }
```
Comment (jeeminso) [inline server/src/main/java/io/crate/metadata/doc/DocTableInfo.java:738]: We already analyze the constraint so I think it should always be false.

**Review by jeeminso — APPROVED** [review]
Just some minor comments, LGTM otherwise

File: server/src/main/java/io/crate/metadata/doc/DocTableInfo.java (hunk 2 above)
Comment (mfussenegger) [reply server/src/main/java/io/crate/metadata/doc/DocTableInfo.java:738]: There could still be a race. The check here is cheap and can avoid a cluster state update which would be expensive compared to the check here

File: server/src/main/java/io/crate/metadata/doc/DocTableInfo.java (hunk 1 above)
Comment (mfussenegger) [reply server/src/main/java/io/crate/metadata/doc/DocTableInfo.java:735]: Yes but the new constraints need to include everything not removed

**Pull request merged**